### PR TITLE
Use Silk icons in forum-toolbar & related changes

### LINKED ISSUE #2595: (Forum material icons) dropdown going over image
### Description

styling has a dropdown  arrow going over the image (just slightly) 
Screenshot 2021-05-24 2 20 33 PM

### Repoduction steps
go to an open forums topic

### Expected behaviour
css to be clean

### Actual behaviour

goes slightly over onto the other pic
### Screenshots

Screenshot 2021-05-24 2 20 33 PM

### Environment

ChromeOS 13904.37.0, Chrome 91.0.4472.66

### Additional context

sorry if this is the wrong type.

### LINKED ISSUE #3653: `forum-toolbar` doesn't account for `material-forum-editor-buttons` being off
## Description

The `forum-toolbar` addon ( `More forum toolbar buttons` ) does not account for `material-forum-editor-buttons` ( `Better forum post editor buttons` ) being turned off.

## Repoduction

1. Turn off `Better forum post editor buttons` and turn on `More forum toolbar buttons`.
2. Find an _open_ forum topic, like [this shop](https://scratch.mit.edu/discuss/topic/550062/#post). <sub><sup>It was just the first on the Requests forum - I have nothing to do with the Box Shop.</sup></sub>
3. Scroll down to the `New reply` section, and compare the buttons added by `forum-toolbar` to the default buttons.

## Expectation

If `material-forum-editor-buttons` is on, the buttons added by `forum-toolbar` should also have material icons; However, if it's off, `forum-toolbar`'s button's icons should _not_ be material, and should instead be in the pixelated style of the vanilla icons.

## Reality

No matter what the status of `material-forum-editor-buttons` is, the buttons added by `forum-toolbar` are _always_ material.

screenshot

## Environment

- Browser: Microsoft Edge 95.0.1020.20 Beta, Firefox 93.0
- Operating system: Windows 10 20H1 x64
- Scratch Addons version: v1.21.0
- Addons JSON:  
```jsonc
{"core":{"lightTheme":true,"version":"1.21.0"},"addons":{"editor-searchable-dropdowns":{"enabled":true,"settings":{}},"editor-devtools":{"enabled":true,"settings":{"enableCleanUpPlus":true,"enablePasteBlocksAtMouse":true,"enableMiddleClickFinder":true}},"forum-search":{"enabled":true,"settings":{}},"msg-count-badge":{"enabled":false,"settings":{"color":"#175ef8","showOffline":true}},"scratch-notifier":{"enabled":true,"settings":{"mark_as_read_when_clicked":true,"notification_sound":"system-default","commentsforme_notifications":true,"commentsonmyprojects_notifications":true,"forumpost_notifications":true,"loveproject_notifications":true,"favoriteproject_notifications":true,"followuser_notifications":true,"curatorinvite_notifications":true,"studioactivity_notifications":false,"remixproject_notifications":true,"becomeownerstudio_notifications":true,"becomehoststudio_notifications":true}},"discuss-button":{"enabled":true,"settings":{"removeIdeasBtn":true,"buttonName":"Discuss"}},"editor-stepping":{"enabled":true,"settings":{"highlight-color":"#0000ff"}},"image-uploader":{"enabled":true,"settings":{"project_thumbnails":false}},"scratch-messaging":{"enabled":false,"settings":{}},"forum-time-zones":{"enabled":true,"settings":{}},"remove-sprite-confirm":{"enabled":false,"settings":{}},"true-youtube-links":{"enabled":true,"settings":{}},"progress-bar":{"enabled":true,"settings":{"topbar":true,"height":5}},"remix-tree-button":{"enabled":true,"settings":{"buttonColor":"#4d97ff"}},"bitmap-copy":{"enabled":true,"settings":{}},"more-links":{"enabled":true,"settings":{}},"youtube-fullscreen":{"enabled":true,"settings":{}},"full-signature":{"enabled":true,"settings":{"whatworkingon":true,"whathappen":true,"signature":true,"blocks":true}},"60fps":{"enabled":true,"settings":{"framerate":60}},"search-profile":{"enabled":true,"settings":{}},"better-emojis":{"enabled":false,"settings":{}},"cloud-games":{"enabled":false,"settings":{}},"studio-tools":{"enabled":true,"settings":{"mystufftools":true}},"last-edit-tooltip":{"enabled":true,"settings":{}},"semicolon":{"enabled":false,"settings":{}},"infinite-scroll":{"enabled":true,"settings":{"forumScroll":true,"profileCommentScroll":true,"projectScroll":true,"studioScroll":true,"studioProjectScroll":true,"studioBrowseProjectScroll":true,"studioCuratorScroll":true,"studioActivityScroll":true,"messageScroll":true,"showFooter":false}},"mouse-pos":{"enabled":true,"settings":{}},"clones":{"enabled":true,"settings":{}},"exact-count":{"enabled":true,"settings":{"user":true,"studio":true,"forumuser":true}},"cat-blocks":{"enabled":true,"settings":{"watch":false}},"editor-messages":{"enabled":true,"settings":{}},"pause":{"enabled":true,"settings":{}},"resizable-comment-input":{"enabled":false,"settings":{}},"animated-thumb":{"enabled":false,"settings":{"persistentThumb":false}},"live-featured-project":{"enabled":true,"settings":{"showMenu":false,"autoPlay":true,"forceAlternative":true,"alternativePlayer":"forkphorus","enableTWAddons":true}},"forum-quote-code-beautifier":{"enabled":true,"settings":{"bordercolor":"#28A5DA"}},"fix-uploaded-svgs":{"enabled":true,"settings":{}},"project-info":{"enabled":true,"settings":{"editorCount":true}},"curator-link":{"enabled":true,"settings":{"styleAsNormalText":true}},"confirm-actions":{"enabled":true,"settings":{"projectsharing":true,"followinguser":false,"joiningstudio":false,"closingtopic":true}},"feature-unshared":{"enabled":true,"settings":{}},"project-notes-tabs":{"enabled":true,"settings":{}},"block-switching":{"enabled":true,"settings":{"noop":true,"border":true,"motion":true,"looks":true,"sound":true,"event":true,"control":true,"operator":true,"sensing":true,"data":true,"extension":true,"customargs":true,"customargsmode":"defOnly","sa":true}},"my-ocular":{"enabled":true,"settings":{"profile":true,"discuss":true,"reactions":true}},"better-featured-project":{"enabled":true,"settings":{"blur":0}},"mediarecorder":{"enabled":true,"settings":{}},"color-picker":{"enabled":true,"settings":{}},"onion-skinning":{"enabled":false,"settings":{"default":false,"previous":1,"next":0,"opacity":25,"opacityStep":10,"layering":"front","mode":"tint","beforeTint":"#FF0000","afterTint":"#0000FF"}},"better-quoter":{"enabled":true,"settings":{}},"data-category-tweaks-v2":{"enabled":true,"settings":{"separateListCategory":true,"separateLocalVariables":true,"moveReportersDown":true}},"mute-project":{"enabled":true,"settings":{}},"disable-auto-save":{"enabled":false,"settings":{}},"hide-flyout":{"enabled":true,"settings":{"toggle":"cathover","speed":"short"}},"comments-linebreaks":{"enabled":true,"settings":{}},"account-settings-capitalize":{"enabled":true,"settings":{}},"longer-wiwo":{"enabled":true,"settings":{}},"show-bbcode":{"enabled":true,"settings":{}},"copy-message-link":{"enabled":true,"settings":{}},"load-extensions":{"enabled":true,"settings":{"music":false,"pen":true,"text2speech":false,"translate":false,"videoSensing":false}},"variable-manager":{"enabled":true,"settings":{}},"scratchstats":{"enabled":true,"settings":{}},"folders":{"enabled":true,"settings":{}},"message-filters":{"enabled":true,"settings":{}},"block-palette-icons":{"enabled":true,"settings":{}},"custom-block-shape":{"enabled":false,"settings":{"paddingSize":100,"cornerSize":200,"notchSize":100}},"2d-color-picker":{"enabled":true,"settings":{}},"drag-drop":{"enabled":true,"settings":{"use-hd-upload":true}},"dango-rain":{"enabled":false,"settings":{}},"initialise-sprite-position":{"enabled":true,"settings":{"x":0,"y":0,"library":false}},"custom-zoom":{"enabled":true,"settings":{"maxZoom":500,"minZoom":40,"startZoom":68,"zoomSpeed":50,"autohide":true,"speed":"short"}},"blocks2image":{"enabled":true,"settings":{}},"editor-sounds":{"enabled":false,"settings":{}},"transparent-orphans":{"enabled":true,"settings":{"block":2,"orphan":50,"dragged":15}},"material-forum-editor-buttons":{"enabled":false,"settings":{}},"gamepad":{"enabled":true,"settings":{"hide":true}},"remove-curved-stage-border":{"enabled":true,"settings":{}},"fix-pasted-scripts":{"enabled":true,"settings":{}},"ctrl-click-run-scripts":{"enabled":false,"settings":{}},"redirect-mobile-forums":{"enabled":false,"settings":{}},"expanding-search-bar":{"enabled":true,"settings":{}},"debugger":{"enabled":true,"settings":{}},"editor-dark-mode":{"enabled":true,"settings":{"page":"#111111","primary":"#4d97ff","highlightText":"#4d97ff","menuBar":"#4d97ff","activeTab":"#202020","tab":"#151515","selector":"#202020","selector2":"#202020","selectorSelection":"#111111","accent":"#151515","input":"#202020","workspace":"#151515","categoryMenu":"#202020","palette":"#202020","fullscreen":"#000000","stageHeader":"#000000","border":"#ffffff0d","dots":true,"textShadow":false,"affectStage":true,"darkComments":false,"darkScrollbars":true,"selectedMode":"3-darker","affectPaper":true}},"editor-colored-context-menus":{"enabled":true,"settings":{}},"editor-stage-left":{"enabled":false,"settings":{}},"editor-theme3":{"enabled":true,"settings":{"motion-color":"#4C97FF","looks-color":"#9966FF","sounds-color":"#CF63CF","events-color":"#FFBF00","control-color":"#FFAB19","sensing-color":"#5CB1D6","operators-color":"#59C059","data-color":"#FF8C1A","data-lists-color":"#FF661A","custom-color":"#5f49d8","Pen-color":"#0FBD8C","input-color":"#202020","text":"colorOnBlack","sa-color":"#29BEB8"}},"scratchr2":{"enabled":true,"settings":{"primaryColor":"#4d97ff","linkColor":"#4d97ff","activeColor":"#4280d7"}},"dark-www":{"enabled":true,"settings":{"selectedMode":"experimental-dark"}},"better-img-uploads":{"enabled":true,"settings":{"fitting":"fit"}},"disable-stage-drag-select":{"enabled":true,"settings":{}},"studio-followers":{"enabled":true,"settings":{}},"old-studio-layout":{"enabled":false,"settings":{"version":"scratchr2"}},"items-per-row":{"enabled":false,"settings":{"search":4,"studioProjects":3,"studioCurators":3,"projects":5,"studios":4,"users":11}},"default-project":{"enabled":true,"settings":{"projectId":558275622}},"turbowarp-player":{"enabled":true,"settings":{"action":"player","addons":true}},"ctrl-enter-post":{"enabled":true,"settings":{"comments":true,"forums":false}},"compact-messages":{"enabled":false,"settings":{"hide_icons":false,"message_padding":5,"comment_padding":5}},"customize-avatar-border":{"enabled":false,"settings":{"hide-outline":false,"fill-transparent":false,"outline-color":"#4d97ff40"}},"forum-toolbar":{"enabled":true,"settings":{"wp":true,"wiki":false,"google":false,"dictionary":false,"color":true,"center":true,"code":true}},"no-script-bumping":{"enabled":true,"settings":{}},"user-id":{"enabled":true,"settings":{}},"hide-delete-button":{"enabled":true,"settings":{"sprites":true,"costumes":true,"sounds":true}},"name-backpack-items":{"enabled":true,"settings":{}},"forum-copy-code":{"enabled":true,"settings":{}},"paint-by-default":{"enabled":true,"settings":{"sprite":"paint","costume":"upload","backdrop":"paint","sound":"upload"}},"hide-project-stats":{"enabled":true,"settings":{"loves":true,"favorites":true,"remixes":true,"views":false,"lovesFront":false,"remixesFront":true}},"scratchblocks":{"enabled":true,"settings":{}},"editor-unshare-button":{"enabled":true,"settings":{}},"remix-button":{"enabled":false,"settings":{}},"move-to-top-bottom":{"enabled":true,"settings":{}},"editor-extra-keys":{"enabled":true,"settings":{}},"block-cherry-picking":{"enabled":true,"settings":{"invertDrag":false}},"hide-new-variables":{"enabled":true,"settings":{}}}}
```

---

I'm not sure if this is exactly a "bug", though...

### LINKED ISSUE #3941: `material-forum-editor-buttons`: buttons look weird
### Description
All of the editor buttons look weird

### Repoduction steps

1. Go to a place on the forums where you can post, like [this](https://scratch.mit.edu/discuss/topic/514769/#reply).

### Expected behaviour
The buttons should look like how they did in 1.22.x (not messed up)

### Actual behaviour
They don't

### Screenshots
Screenshot 2021-12-17 140829

### Environment

- Browser:  Edge 96.0.1054.57
- Scratch Addons version:  1.23.0
- Operating system:  Windows 10 x64

### PR DESCRIPTION
**Pull request by mxmou**
Resolves #2595
Resolves #3653
Resolves #3941

### Changes

* Replaces Material icons added by `forum-toolbar` with icons from the icon set used by markitup if `material-forum-editor-buttons` is disabled
image
* Changes to `material-forum-editor-buttons`:
  * Adds a Material dropdown icon - this was in the original userstyle, but it was somehow lost when it was converted to an addon
  * Now looks better in dark mode: 
  image
  * Removes unnecessary repetition of `background-size: 20px;` and unused CSS
  * Replaces the clean icon with a more appropriate one

### CODE REVIEW

**Comment by Secret-chest** [comment]
Thanks!

**Comment by Mr-Anonymous75** [comment]
Wouldn't the [`color_swatch.png`](http://www.famfamfam.com/lab/icons/silk/icons/color_swatch.png) icon you found yourself be a better icon for the color button?

**Comment by mxmou** [comment]
> Wouldn't the [`color_swatch.png`](http://www.famfamfam.com/lab/icons/silk/icons/color_swatch.png) icon you found yourself be a better icon for the color button?

I used the palette instead because it's more similar to the previous icon.

**Comment by Mr-Anonymous75** [comment]
> > Wouldn't the [`color_swatch.png`](http://www.famfamfam.com/lab/icons/silk/icons/color_swatch.png) icon you found yourself be a better icon for the color button?
> 
> I used the palette instead because it's more similar to the previous icon.

But it looks kinda out of place, with that brown background.

File: addons/forum-toolbar/addon.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "More forum toolbar buttons",
-  "description": "Adds buttons in the forum editing toolbar to insert BBCode tags, such as [center] or [color].",
+  "description": "Adds buttons in the forum editing toolbar to insert BBCode tags, such as [center] or [color]. Uses icons from the Silk icon set (http://www.famfamfam.com/lab/icons/silk/).",
```
Comment (WorldLanguages) [inline addons/forum-toolbar/addon.json:3]: Why not credit Silk instead of using the description?

**Pull request merged**